### DISCUSSION TITLE
Version 2: replace webpack with AssetMapper / Symfony 6.4 / 7

### DISCUSSION THREAD

**Discussion by tacman** [Part-DB/Part-DB-server#404] [General]
First, kudos -- what a great application!  I ran across it when looking for examples of Symfony 6 projects that adhere to some Best Practices, and this one certainly does.   

I'm eagerly awaiting Symfony 6.4 / 7, especially because with the new AssetMapper component, we can drop the build step.  And fortunately, this application does not have a lot of Javascript libraries, and already integrates stimulus.  So it should be a good candidate when the time comes.

I keep running into issues using both AssetMapper (which is experimental in Symfony 6.3) with webpack.  But that might be a first step.

**Comment by jbtronics** [comment]
Webpack does some things in the build step, which is not possible assetmapper, like minifying JS and CSS files, doing tree shaking to reduce the size by a lot and building pre-compressed versions of the assets. Also CKeditor needs a webpack plugin to build its assets, so I dont think its really possible to (easily) replace webpack with assetmapper.

Also when using a docker image (what a major part of the user base does), then the build step is not necessary as this is already done in the docker image build. And if you cannot or want not to install yarn on your server to build the assets in a direct installation, you can download the prebuilt assets, which are automatically build by Github actions for each commits (https://github.com/Part-DB/Part-DB-server/actions/workflows/assets_artifact_build.yml) and which are also included on the releases page.
